app/models/concerns/user_concern.rb

```diff
@@ -60,7 +60,7 @@ module UserConcern
     after_save :create_access_group
     # END_ACL
 
-    validates :email, presence: true, uniqueness: true, email_format: { check_mx: true }, length: { maximum: 250 }, on: :update
+    validates :email, presence: true, uniqueness: true, email_format: { check_mx: true }, length: { maximum: 250 }
```

Comment:
@eanders - wanted to call out this change in behavior. It seems unexpected that we are not validating the email when the user is created. If this seems to risky we can make it conditional using an ephemeral attribute on user to control it
```ruby
@user.ensure_email_is_validated = true
@user.save
```

Reply:
@gigxz do you know if we ever create users without a valid email intentionally?  I think maybe for OKTA integrations, or imports?  I agree with @ttoomey we should be validating on create, but maybe that's not helpful in some scenarios.

Reply:
No, not that I know of. For both Okta and import scenarios, I think it makes sense to validate the email right away. The only different on account creation for Okta is that we can skip sending the initial email invitation (if "Skip email invitation" checked). The email still needs to be entered initially and should be valid.

I see that on the AccountRequest model, there is a similar validation for email that is validated on `create`. So maybe this is already something we validate in a different place? I don't know how the account request model is used though. https://github.com/greenriver/hmis-warehouse/blob/release-148/app/models/account_request.rb#L11

Reply:
> The only different on account creation for Okta is that we can skip sending the initial email invitation

I don't think this validation would impact sending an invitation. I could be wrong but it seems like invitations are sent by manually calling `User.invite!` - we have a different path for user creation in `OmniauthUserProvisioner`

In fact, the omniauth provisioner skips all validations, which would include this email validation.

>  I don't know how the account request model is used though.

Good find; I didn't realize that we support account requests. I wonder if it's still used?